packages/ember-runtime/lib/system/array_proxy.js

```diff
@@ -230,6 +230,9 @@ export default class ArrayProxy extends EmberObject {
 
       this.arrayContentDidChange(0, oldLength, newLength);
       this._addArrangedContentArrayObsever();
+    } else if (key === 'content') {
```

Comment:
at this point we should extract into an invalidate method

Reply:
Updated to use a shared `_invalidate` method.